test/FunctionalTests/Microsoft.OData.Edm.Tests/Csdl/CsdlWriterTests.cs

```diff
@@ -961,6 +961,40 @@ public void ShouldWriteAnnotationForEnumMember()
             Assert.Equal(expected, csdlStr);
         }
 
+        [Fact]
+        public void CanWritePropertyWithCoreTypeDefinitionAndValidationPassed()
+        {
+            string expected =
+            "<?xml version=\"1.0\" encoding=\"utf-16\"?>" +
+            "<edmx:Edmx Version=\"4.0\" xmlns:edmx=\"http://docs.oasis-open.org/odata/ns/edmx\">" +
+              "<edmx:DataServices>" +
+                "<Schema Namespace=\"NS\" xmlns=\"http://docs.oasis-open.org/odata/ns/edm\">" +
+                  "<ComplexType Name=\"Complex\">" +
+                    "<Property Name=\"ModifiedDate\" Type=\"Org.OData.Core.V1.LocalDateTime\" />" +
+                    "<Property Name=\"QualifiedName\" Type=\"Org.OData.Core.V1.QualifiedTypeName\" />" +
+                  "</ComplexType>" +
+                "</Schema>" +
+              "</edmx:DataServices>" +
+            "</edmx:Edmx>";
+
+            EdmModel model = new EdmModel();
+            var localDateTime = model.FindType("Org.OData.Core.V1.LocalDateTime") as IEdmTypeDefinition;
+            Assert.NotNull(localDateTime);
+
+            var qualifiedTypeName = model.FindType("Org.OData.Core.V1.QualifiedTypeName") as IEdmTypeDefinition;
+            Assert.NotNull(qualifiedTypeName);
+
+            EdmComplexType type = new EdmComplexType("NS", "Complex");
+            type.AddStructuralProperty("ModifiedDate", new EdmTypeDefinitionReference(localDateTime, true));
+            type.AddStructuralProperty("QualifiedName", new EdmTypeDefinitionReference(qualifiedTypeName, true));
+
+            model.AddElement(type);
+            IEnumerable<EdmError> errors;
+            Assert.True(model.Validate(out errors));

```

Comment:
should you add an Assert.Equal(0,errors.Count())?

Reply:
Thanks. Normally, it's unnecessary.
 `Validate` method just likes "Try and Do something". Return `true` means there's no errors.

In the inner, this `validate` method is returning using the code like

 `return errors.FirstOrDefault() == null;`